mlflow/server/js/src/utils/Utils.test.js

```diff
@@ -153,7 +153,19 @@ test("formatSource & renderSource", () => {
   };
   const wrapper2 = shallow(Utils.renderSource(databricksRun, databricksRunRevisionTags));
   expect(wrapper2.is("a")).toEqual(true);
-  expect(wrapper2.props().href).toEqual("../#notebook/13/revision/42");
+  expect(wrapper2.props().href).toEqual("http://localhost/#notebook/13/revision/42");
+
+  const wrapper3 = shallow(Utils.renderSource(databricksRun, databricksRunRevisionTags, "?o=123"));
+  expect(wrapper3.is("a")).toEqual(true);
+  // Query params must appear before the hash, see https://tools.ietf.org/html/rfc3986#section-4.2
+  // and https://stackoverflow.com/a/34772568
+  expect(wrapper3.props().href).toEqual("http://localhost/?o=123#notebook/13/revision/42");
+});
+
+test("addQueryParams", () => {
+  expect(Utils.addQueryParams("http://localhost/foo", "?o=123")).toEqual("http://localhost/foo?o=123");
```

Comment:
I think in the future, we should aim to use these.. 
https://github.com/producthunt/chai-enzyme